Dockerfile

```
@@ -23,16 +23,86 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
       libpq-dev \
     && rm -rf /var/lib/apt/lists/*
 
-# CUDA
-WORKDIR /tmp/cuda
-RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/cuda-repo-ubuntu1604_10.0.130-1_amd64.deb \
-    && wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/7fa2af80.pub \
-    && apt-key add 7fa2af80.pub \
-    && apt install ./cuda-repo-ubuntu1604_10.0.130-1_amd64.deb \
-    && apt-get update && apt-get install -y --no-install-recommends \
-      cuda-10-0 \
-    && rm -rf /var/lib/apt/lists/* \
-    && rm -rf /tmp/cuda
+# CUDA base
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    ca-certificates apt-transport-https gnupg-curl && \
+    NVIDIA_GPGKEY_SUM=d1be581509378368edeec8c1eb2958702feedf3bc3d17011adbf24efacce4ab5 && \
+    NVIDIA_GPGKEY_FPR=ae09fe4bbd223a84b2ccfce3f60f4b3d7fa2af80 && \
+    apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/7fa2af80.pub && \
+    apt-key adv --export --no-emit-version -a $NVIDIA_GPGKEY_FPR | tail -n +5 > cudasign.pub && \
+    echo "$NVIDIA_GPGKEY_SUM  cudasign.pub" | sha256sum -c --strict - && rm cudasign.pub && \
```

I like using curly braces around shell variables. I find it to be more readable and it is universal across shells and Makefiles. Some things (ex: Makefile) interpret $blah as $b+lah.
```suggestion
    echo "${NVIDIA_GPGKEY_SUM}  cudasign.pub" | sha256sum -c --strict - && rm cudasign.pub && \
```

Same for the `CUDA_PKG_` references later.

---

fyi I copied this verbatim from nvidia's official docker images. would you recommend changing or can I keep as-is?